### PR TITLE
Build for x86_64-w64-mingw32 target

### PR DESCRIPTION
**Pull request by Blackhex**
Add build for x86_64-w64-mingw32 target using GHA matrix feature.

### CODE REVIEW

**Comment by ZacWalk** [comment]
This looks good for me. I had something similar locally but this is better.

**Review by ZacWalk — APPROVED** [review]

**Comment by ZacWalk** [comment]
I tried building this on my machine and it failed. Best not to merge before the build is fixed.

File: build.sh [hunk 1]
```diff
@@ -286,13 +297,13 @@ fi
 
 if [ $RUN_CONFIG = 1 ] ; then
     config_binutils
-    config_gcc_compiler
     config_mingw_headers
```
Comment (Blackhex) [inline build.sh:300]: I belive the headers should be built first (on a clean environment).

**Comment by Blackhex** [comment]
> I tried building this on my machine and it failed. Best not to merge before the build is fixed.

What was the error? Now it passes on the CI (https://github.com/ZacWalk/mingw-woarm64-build/actions/runs/6721553244) and even on my machine.

**Comment by Blackhex** [comment]
OK, let me separate it into multiple commits.

**Comment by ZacWalk** [comment]
Also, how do you run this locally? Do you need to set a variable for TARGET_ARCH? It would be good to have it default to TARGET_ARCH=aarch64 if nothing was set externally.

**Comment by Blackhex** [comment]
> Also, how do you run this locally? Do you need to set a variable for TARGET_ARCH? It would be good to have it default to TARGET_ARCH=aarch64 if nothing was set externally.

The default is `aarch64`, see https://github.com/ZacWalk/mingw-woarm64-build/pull/25/files#diff-4d2a8eefdf2a9783512a35da4dc7676a66404b6f3826a8af9aad038722da6823R26, I can add script parameters handling to allow users to override this other way than with the env variable if that is what you mean?

**Comment by Blackhex** [comment]
Deprecated in favor of https://github.com/ZacWalk/mingw-woarm64-build/pull/40

**Pull request closed without merging**